app/controllers/movies_controller.rb

```diff
@@ -0,0 +1,82 @@
+class MoviesController < ApplicationController
+  include ApplicationHelper
+
+  def index
+    render 'index'
+  end
+
+
+  def show
+    @movie = Movie.find_by(id: params[:id])
+    @comments = @movie.comments
+    @conversation = Conversation.new
+    activity_logger('movie', @movie) if @movie
+    @comment = Comment.new
+    @related_movies = @movie.related_movies
+
+  end
+
+  def search
+    if request.xhr?
+      @movies = get_movies_by_title(params["searchResult"])
+    else
+      @movies = get_movies_by_title(params['search_form'])
+    end
+
+    if @movies
+      @movies.map! do |movie|
+        movie = Movie.find_by(imdbID: movie.imdbID)
+      end
+    end
+
+
+    if @movies && !request.xhr?
+      activity_logger('search', params['search_form'])
+      render "movies/index"
+    elsif request.xhr?
+      render :json => @movies
+    else
+      @errors = 'Movie not found' # need to implement error message partials
+      render 'movies/index'
+    end
+
+  end
+
+  def like
+    require_user
+    @movie = find_movie(params[:id])
+    like = @movie.likes.create(user_id: session[:user_id])
+
+    activity_logger('like', @movie) if like.valid?
+    redirect_to :back
```

Comment:
From a UX standpoint, liking a movie is a good place to implement asychronous HTTP requests on the front end, instead of hitting a redirect on the server. (Your users are not going to expect the page to reload when they click on a like button.) (Note: same with unliking a movie.)